### PR TITLE
[ISSUE-4063]The material package removes the version prefix 'v'

### PR DESCRIPTION
**Pull request by binbinCheng**
### What is the purpose of the change

**Remove the 'v prefix of the material package name when the material package is packaged and installed**

### Related issues/PRs

Related issues: #4063
Related pr:#4064

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

**Comment by casionone** [comment]
function getEngineConnBMLResources need update

**Review by GuoPhilipse — CHANGES_REQUESTED** [review]
same here.
linkis-engineconn-plugins/shell/src/main/assembly/distribution.xml
		
        <fileSet>
            <directory>${basedir}/src/main/resources/bin</directory>
            <includes>
                <include>*</include>
            </includes>
            <fileMode>0777</fileMode>
            <outputDirectory>dist/v${shell.version}/bin</outputDirectory>
            <lineEnding>unix</lineEnding>
        </fileSet>

File: linkis-dist/package/sbin/ext/linkis-mg-eureka [hunk 1]
```diff
@@ -84,6 +84,9 @@ SERVER_IP="`hostname --fqdn`"
 if test -z "$EUREKA_INSTALL_IP"
 then
   export EUREKA_INSTALL_IP=$SERVER_IP
+
+
+
 fi
```
Comment (casionone) [inline linkis-dist/package/sbin/ext/linkis-mg-eureka:90]: Remove unused modify

File: linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/scala/org/apache/linkis/engineplugin/server/localize/AbstractEngineConnBmlResourceGenerator.scala [hunk 2]
```diff
@@ -85,15 +86,13 @@ abstract class AbstractEngineConnBmlResourceGenerator extends EngineConnBmlResou
...
         MessageFormat.format(DIST_IS_EMPTY.getErrorDesc, engineConnType)
       )
-    } else if (!children.exists(_.getName.startsWith("v"))) {
-      Array(engineConnDistHome)
-    } else if (children.forall(_.getName.startsWith("v"))) {
-      children.map(_.getPath)
-    } else {
+    } else if (children.apply(0).getName.contains("-")) {
       throw new EngineConnPluginErrorException(
-        DIST_IRREGULAR_EXIST.getErrorCode,
```
Comment (casionone) [inline linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/scala/org/apache/linkis/engineplugin/server/localize/AbstractEngineConnBmlResourceGenerator.scala:94]:
```
├── hive
│   ├── dist
│   │   │── v2.3.3
│   │   │── 2.3.4
│   │   └── 2.3.5
```
Consider the compatibility of multiple versions and old models

考虑多版本和老模式的兼容性

File: linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/scala/org/apache/linkis/engineplugin/server/localize/DefaultEngineConnBmlResourceGenerator.scala [hunk 3]
```diff
@@ -31,10 +33,16 @@ class DefaultEngineConnBmlResourceGenerator
 
   override def generate(engineConnType: String): Map[String, Array[EngineConnLocalizeResource]] =
     getEngineConnDistHomeList(engineConnType).map { path =>
-      val distFile = new File(path)
-      val key = if (distFile.getName.startsWith("v")) distFile.getName else NO_VERSION_MARK
+      val versionFile = new File(path)
```
Comment (casionone) [inline linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/scala/org/apache/linkis/engineplugin/server/localize/DefaultEngineConnBmlResourceGenerator.scala:36]: add log about path and enginetype  info

**Review by casionone — APPROVED** [review]

**Pull request merged**